src/gui/widgets/EffectView.cpp

```diff
@@ -110,17 +111,19 @@ EffectView::EffectView( Effect * _model, QWidget * _parent ) :
 		m_controlView = effect()->controls()->createView();
 		if( m_controlView )
 		{
-			m_subWindow = gui->mainWindow()->addWindowedWidget(
-								m_controlView,
-				Qt::SubWindow | Qt::CustomizeWindowHint  |
-					Qt::WindowTitleHint | Qt::WindowSystemMenuHint );
-			m_subWindow->setSizePolicy( QSizePolicy::Fixed, QSizePolicy::Fixed );
-			m_subWindow->setFixedSize( m_subWindow->size() );
-
-			connect( m_controlView, SIGNAL( closed() ),
-					this, SLOT( closeEffects() ) );
-
-			m_subWindow->hide();
+			if( !(m_controlView->windowTitle() == "Peak Controller") )
```

Comment:
> more elegant way to identify the peak controller

Research points to [`dynamic_cast` against `PeakController`  and checking for `!= NULL`](http://stackoverflow.com/a/351865/3196753)